### PR TITLE
PureScript 0.12 support

### LINKED ISSUE #90: Support PureSctript 0.12

### PR DESCRIPTION
**Pull request by akheron**
- Upgrade all dependencies
- Use Effect instead of Eff
- Use Foreign.Object instead of StrMap
- Use Effect.Ref instead of Control.Monad.ST
- Drop SecureServerM, it's the same as ServerM now

Fixes #90

### CODE REVIEW

File: src/HTTPure/Headers.purs [hunk 1]
```diff
@@ -9,62 +9,61 @@ module HTTPure.Headers
 
 import Prelude
 
-import Control.Monad.Eff as Eff
-import Data.String as StringUtil
-import Data.StrMap as StrMap
+import Effect as Effect
+import Foreign.Object as Object
+import Data.String as String
```
[the remark this answers is not in the document]
Comment (cprussin) [reply src/HTTPure/Headers.purs]: I went ahead and merged this PR because I'd like to go ahead and release.  I'll likely revisit this later.

**Comment by akheron** [comment]
`Effect` is the untagged effect monad that's analogous to Haskell's `IO`. Some insight to why the name `Effect` was chosen can be found [here](https://github.com/purescript/purescript-eff/issues/25).

**Comment by akheron** [comment]
And thank you for HTTPure! It seems there aren't so many HTTP servers for PureScript, and I like your approach of pure requests and responses. Clean code and unit tests made it easy to upgrade to 0.12 :)

**Comment by cprussin** [comment]
@akheron Thanks for the context.  That's interesting; so unless I'm misreading it sounds like the path forward should be to keep `Effect` for what you've done here, and to (eventually) move `Aff` over to `IO` instead.

In any case, I've released v0.7.1.  Thanks again for the contribution!

**Pull request merged**